File: client/go/src/example/quick/quickStart.go
package main

import (
	"os"
	"fmt"
	
	"example/client"
	"example/connect"
	"example/device"
	"example/user"
	"example/finger"
	"example/card"
	"example/event"
)

const (
	CA_FILE = "../../../../cert/ca.crt"
	SERVER_IP = "192.168.0.2"
	SERVER_PORT = 4000
	
	A2_IP = "192.168.0.110"
	A2_PORT = 51211
	USE_SSL = false
)

var (
	grpcClient *client.GrpcClient

	connectSvc *connect.ConnectSvc
	deviceSvc *device.DeviceSvc
	userSvc *user.UserSvc
	fingerSvc *finger.FingerSvc
	cardSvc *card.CardSvc
	eventSvc *event.EventSvc
)


func testConnect() (uint32, error) {
	devList, err := connectSvc.GetDeviceList()

	if err != nil {
		return 0, err
	}

	fmt.Printf("Device list before connection: %v\n\n", devList)

	deviceID, err := connectSvc.Connect(A2_IP, A2_PORT, USE_SSL)

	if err != nil {
		return 0, err
	}	

	devList, err = connectSvc.GetDeviceList()

	if err != nil {
		return 0, err
	}

	fmt.Printf("Device list after connection %v\n\n", devList)

	return deviceID, nil
}


func testDevice(deviceID uint32) error {
	devInfo, err := deviceSvc.GetInfo(deviceID)

	if err != nil {
		return err
	}

	fmt.Printf("Device info: %v\n\n", devInfo)

	capInfo, err := deviceSvc.GetCapabilityInfo(deviceID)

	if err != nil {
		return err
	}

	fmt.Printf("Device capability info: %v\n\n", capInfo)

	return nil
}


func main() {
	grpcClient = client.NewGrpcClient()

	err := grpcClient.Connect(CA_FILE, SERVER_IP, SERVER_PORT)
	if err != nil {
		os.Exit(1)
	}

	connectSvc = connect.NewConnectSvc(grpcClient.GetConn())
	deviceSvc = device.NewDeviceSvc(grpcClient.GetConn())
	userSvc = user.NewUserSvc(grpcClient.GetConn())
	fingerSvc = finger.NewFingerSvc(grpcClient.GetConn())
	cardSvc = card.NewCardSvc(grpcClient.GetConn())
	eventSvc = event.NewEventSvc(grpcClient.GetConn())

	deviceID, err := testConnect()

	if err != nil {
		fmt.Printf("Cannot connect to the device: %v\n", err)
		os.Exit(1)
	}

	testDevice(deviceID)
	
	testFinger(deviceID)
	testCard(deviceID) 

	testUser(deviceID)

	testEvent(deviceID) 

	connectSvc.Disconnect([]uint32{ deviceID })

	grpcClient.Close()

	return
}

File: client/java/src/main/java/com/supremainc/sdk/example/quick/QuickStart.java
package com.supremainc.sdk.example.quick;

import java.util.List;

import com.supremainc.sdk.card.CardGrpc;
import com.supremainc.sdk.connect.ConnectGrpc;
import com.supremainc.sdk.connect.DeviceInfo;
import com.supremainc.sdk.connect.ConnectInfo;
import com.supremainc.sdk.device.CapabilityInfo;
import com.supremainc.sdk.device.DeviceGrpc;
import com.supremainc.sdk.device.FactoryInfo;
import com.supremainc.sdk.event.EventGrpc;
import com.supremainc.sdk.example.card.CardSvc;
import com.supremainc.sdk.example.client.GrpcClient;
import com.supremainc.sdk.example.connect.ConnectSvc;
import com.supremainc.sdk.example.device.DeviceSvc;
import com.supremainc.sdk.example.event.EventSvc;
import com.supremainc.sdk.example.finger.FingerSvc;
import com.supremainc.sdk.example.user.UserSvc;
import com.supremainc.sdk.finger.FingerGrpc;
import com.supremainc.sdk.user.UserGrpc;

public class QuickStart {
  private static final String CA_FILE = "ca.crt";

  private static final String SERVER_ADDR = "192.168.0.2";
  private static final int SERVER_PORT = 4000;

  private static final String DEVICE_ADDR = "192.168.0.110";
  private static final int DEVICE_PORT = 51211;
  private static final boolean DEVICE_USE_SSL = false;

  private GrpcClient grpcClient;
  private ConnectSvc connectSvc;
  private DeviceSvc deviceSvc;
  private UserSvc userSvc;
  private FingerSvc fingerSvc;
  private CardSvc cardSvc;
  private EventSvc eventSvc;

  public QuickStart(GrpcClient client) {
    grpcClient = client;

    connectSvc = new ConnectSvc(ConnectGrpc.newBlockingStub(client.getChannel())); 
    deviceSvc = new DeviceSvc(DeviceGrpc.newBlockingStub(client.getChannel()));   
    userSvc = new UserSvc(UserGrpc.newBlockingStub(client.getChannel()));  
    fingerSvc = new FingerSvc(FingerGrpc.newBlockingStub(client.getChannel()));  
    cardSvc = new CardSvc(CardGrpc.newBlockingStub(client.getChannel()));
    eventSvc = new EventSvc(EventGrpc.newBlockingStub(client.getChannel()));
  }

  private int testConnect() throws Exception {
    List<DeviceInfo> devInfo = connectSvc.getDeviceList();

    System.out.printf("Device list before connection: \n%s\n\n", devInfo);

    ConnectInfo connInfo = ConnectInfo.newBuilder().setIPAddr(DEVICE_ADDR).setPort(DEVICE_PORT).setUseSSL(DEVICE_USE_SSL).build();

    int deviceID = connectSvc.connect(connInfo);

    devInfo = connectSvc.getDeviceList();

    System.out.printf("Device list after connection: \n%s\n\n", devInfo);

    return deviceID;
  }

  private void testDevice(int deviceID) throws Exception {
    FactoryInfo factoryInfo = deviceSvc.getInfo(deviceID);

    System.out.printf("Device info: \n%s\n\n", factoryInfo);    

    CapabilityInfo capinfo = deviceSvc.getCapabilityInfo(deviceID);

    System.out.printf("Device capability info: \n%s\n\n", capinfo);    
  }


  public static void main(String[] args) throws Exception {
    GrpcClient client = new GrpcClient();

    try {
      client.connect(CA_FILE, SERVER_ADDR, SERVER_PORT);
    } catch (Exception e) {
      System.out.printf("Cannot connect to the server: %s", e); 
      System.exit(-1);
    }

    try {
      QuickStart quickStart = new QuickStart(client);

      int deviceID = quickStart.testConnect();

      quickStart.testDevice(deviceID);

      new FingerTest(quickStart.fingerSvc).test(deviceID);
      new CardTest(quickStart.cardSvc).test(deviceID); 
      new UserTest(quickStart.userSvc, quickStart.fingerSvc).test(deviceID);
      new EventTest(quickStart.eventSvc).test(deviceID);

      quickStart.connectSvc.disconnect(deviceID);      
      client.close();
    } catch (Exception e) {
      System.out.printf("Cannot complete the test: %s", e); 
    }
  }
}

